### PR TITLE
use emotion

### PR DESCRIPTION
**Pull request by alxnddr**
### Changes

0) Remove styled-components and its loader dependencies
1) Replace imports
2) Add `emotion` babel loader
3) Remove `forwardRefToInnerRef` and replace `innerRef`s with just `ref`s. For class component I renamed it to `forwardedRef`
4) Add and use `data-testids` to fix spec relied on class names generated by styled-component
5) Fix props types: from `styled<Props>(MyComponent)...` to `styled(MyComponent)<Props>...` 
6) Fix typos in CSS that did not break it with SC but broke with emotion
7) Added types to some forwardRefs
8) Had to add a bunch of `.click({ force: true })` in Cypress specs for accordion lists in popovers
9) Change `css` interpolation with props 

**Before**
```typescript
const hoverStyle = css`
    color: ${props => color(props.activeColor)};
`
```

**After**
```typescript
const hoverStyle = props => css`
    color: ${color(props.activeColor)};
`
```

### CODE REVIEW

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

File: frontend/src/metabase/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -54,7 +57,8 @@ function TextInput({
     <TextInputRoot className={className}>
       {icon && <IconWrapper>{icon}</IconWrapper>}
       <Input
-        innerRef={innerRef}
+        as={as}
```
Comment (ranquild) [inline frontend/src/metabase/components/TextInput/TextInput.tsx:60]: Why do we need `as` here?

**Review by ranquild — APPROVED** [review]
Works correctly. I'm going to spend some time to figure out what causes us to use `force: true`, but that shouldn't block the PR

**Comment by alxnddr** [comment]
> Works correctly. I'm going to spend some time to figure out what causes us to use `force: true`, but that shouldn't block the PR

I and @nemanjaglumac looked at it yesterday in the context of Select keyboard nav PR. Basically, there is no JS event on the element, but Cypress claims it clicked on an element. I assume popover quickly changes its position after cypress calculated coordinates or something like that 🤯

File: frontend/src/metabase/components/TextInput/TextInput.tsx (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/components/TextInput/TextInput.tsx]: Good catch, I removed this accidental `as`, thanks

**Pull request merged**